frontend/src/metabase/components/CopyButton.jsx

```diff
@@ -35,7 +35,7 @@ export default class CopyWidget extends Component {
     return (
       <Tooltip tooltip={t`Copied!`} isOpen={this.state.copied}>
         <CopyToClipboard text={value} onCopy={this.onCopy}>
-          <div className={className} style={style}>
+          <div className={className} style={style} data-testid="copy-button">
             <Icon name="copy" {...props} />
```

Comment:
@paulrosenzweig - sorry to bump this but couldn't think of a better way to reach you.

Anyway, I think I've found a buletproof way of targetting icons and the best part is - we don't even have to reinvent the wheel! 🤓

There is literary one line of code we should add to `icon_paths.js` => add `role: "img` to attributes et voilà!

1. it improves accessibility of SVGs ([MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Role_Img)) without cluttering the code
2. we can target icons with `getByRole("img")` which is preferred way of doing things in RTL, anyway
3. there is usually a need to test just one icon in a component - I cannot think of a reason why we should test anything more than whether **an** icon was rendered or not
4. I would expand `Icon.unit.spec.js` to make sure that appropriate icon renders given the `name`, which would give us the confidence neccessary to skip testing that part in other components

#### Bonus 🍰
There is an assertion `toHaveClass()` in [`@testing-library/jest-dom/extend-expect`](https://github.com/testing-library/jest-dom/tree/master/src) for the cases where we need to make absolutely sure that certain icon was loaded (but given the point 4 above, I cannot think of such case)

Hopefully this wasn't confusing. I am a bit excited right now, but I'll sum it up nicely in an issue and open PR soon after that.

Reply:
Sweet! If setting the role is a a11y win and helps write some tests, we should definitely go for it.

> there is usually a need to test just one icon in a component

I do think we'll have situations where we want to distinguish between multiple icons when selecting rather than just asserting which is there. For example, here's some UI from an upcoming feature in 0.37:

![image](https://user-images.githubusercontent.com/691495/92617162-0071f400-f28d-11ea-87bd-a26374aa5b26.png)

I'm not sure whether this specific component will need a unit test touching those buttons, but in general I feel like we use icons in our UI as though they were text and should be able to target them similarly in our testing tools. e.g. saying "click 'Done'" and "click '.Icon-close'" feel similar.

Reply:
Then maybe adding title and description could narrow it down to a particular icon. 
Like we already agreed, having a querySelector here and there won't hurt anyone. I'm certainly thinking about it and trying to find a better solution. In the meantime simply adding role goes a long way.

Btw, I found that `DataSelector.unit.spec.js` targets specific icons. I'll check if there is a some text that could be clicked on instead. So far, the only remaining challenge remains when there is absolutely nothing else - just a pure icon/button.